src/me/remie/vulcan/leaguetasks/LeagueScript.java

```diff
@@ -109,6 +109,19 @@ private void checkTasks() {
             }, 1000, 5);
             return;
         }
+
+        // Make sure we have correct filter active.
+        final SimpleWidget dropdownMenuWidget = ctx.widgets.getWidget(657,31).getChild(4);
+        if (dropdownMenuWidget != null && !dropdownMenuWidget.isHidden() && !dropdownMenuWidget.getText().equalsIgnoreCase("all")) {
```

Comment:
If the leagues interface is open, I don't think this widget would ever be hidden, so we can remove this check.
```suggestion
        if (dropdownMenuWidget != null && !dropdownMenuWidget.getText().equalsIgnoreCase("all")) {
```